node/src/accountant/db_access_objects/receivable_dao.rs

```diff
@@ -118,35 +120,41 @@ impl ReceivableDao for ReceivableDaoReal {
         amount: u128,
     ) -> Result<(), ReceivableDaoError> {
         let main_sql = "insert into receivable (wallet_address, balance_high_b, balance_low_b, last_received_timestamp) values \
-        (:wallet, :balance_high_b, :balance_low_b, :last_received) on conflict (wallet_address) do update set \
+        (:wallet, :balance_high_b, :balance_low_b, :last_received_timestamp) on conflict (wallet_address) do update set \
         balance_high_b = balance_high_b + :balance_high_b, balance_low_b = balance_low_b + :balance_low_b";
         let overflow_update_clause = "update receivable set balance_high_b = :balance_high_b, balance_low_b = :balance_low_b \
         where wallet_address = :wallet";
 
-        let last_received = to_time_t(timestamp);
+        let last_received_timestamp = to_time_t(timestamp);
         let params = SQLParamsBuilder::default()
             .key(WalletAddress(wallet))
             .wei_change(Addition("balance", amount))
             .other_params(vec![Param::MainClauseLimited((
-                ":last_received",
-                &last_received,
+                ":last_received_timestamp",
+                &last_received_timestamp,
             ))])
             .build();
 
         Ok(self.big_int_db_processor.execute(
-            self.conn.as_ref(),
+            Either::Left(self.conn.as_ref()),
             BigIntSqlConfig::new(main_sql, overflow_update_clause, params),
         )?)
     }
 
     fn more_money_received(&mut self, timestamp: SystemTime, payments: Vec<BlockchainTransaction>) {
         Self::multi_row_update_from_received_payments(
-            self.conn.as_mut(),
-            &self.big_int_db_processor,
+            &mut *self.conn,
+            &*self.big_int_db_processor,
             &self.logger,
             timestamp,
             &payments,
         )
+        .unwrap_or_else(|e| {
```

Comment:
If this fails to record incoming payments, I think it should either panic (if the situation is unrecoverable) or return an error value (if the caller might be able to recover it).